### DISCUSSION TITLE
`default_headers` plugin silently fails w/ capitalized `Content-Type` header on Rack 3

### DISCUSSION THREAD

**Discussion by benalavi** [jeremyevans/roda#373] [General]
Ran into this when upgrading to Rack 3. The below app will silently ignore the `Content-Type` header (it returns `content-type: text/html` rather than `content-type: application/json`), but lowercase other arbitrary capitalized headers.

```rb
require "roda"

class API < Roda
  plugin :default_headers,
    "Content-Type" => "application/json",
    "Foo" => "bar"

  route do |r|
    r.get "" do
      "Hello"
    end
  end
end

run API.freeze
```

```
< HTTP/1.1 200 OK
< content-type: text/html
< foo: bar
< Content-Length: 5
```

I assume this is related to the Rack 3 change that all headers must be lowercase. So maybe w/ Rack 3 this is now expected behavior, in which case it might just warrant a tiny change to the documentation: https://github.com/jeremyevans/roda/blob/master/lib/roda/plugins/default_headers.rb#L16 -- but it was surprising that it worked for the `"Foo"` header.

Rack 3 seems to raise now if you try to assign a non-lowercase header, which was helpful for us when upgrading.

Lowercase `content-type` works as expected:

```rb
require "roda"

class API < Roda
  plugin :default_headers,
    "content-type" => "application/json",
    "Foo" => "bar"

  route do |r|
    r.get "" do
      "Hello"
    end
  end
end

run API.freeze
```

```
< HTTP/1.1 200 OK
< content-type: application/json
< foo: bar
< Content-Length: 5
```

**Comment by jeremyevans** [comment]
Thank you very much for the report.  This is because `Roda::RodaResponse::DEFAULT_HEADERS` is not a `Rack::Headers instance` by default.  I'll fix that shortly.

**Comment by jeremyevans** [comment]
Fixed in 8757ac770271a2bd8f81b6fb72706c414da44a8b

**Reply by benalavi** [reply]
Thanks! That fixed our broken case as well 👍
